### PR TITLE
feat(openrouter): mirror workspace guardrails into org policy

### PR DESCRIPTION
**Pull request by chaliy**
## What

Adds `OpenRouterWorkspaceCapability` with two host-facing tools that let an agent inspect upstream OpenRouter workspace constraints and detect incompatibilities before committing to a routing configuration.

New file: `crates/core/src/capabilities/openrouter_workspace.rs`

## Why

Everruns can send requests through OpenRouter but had no way to inspect upstream workspace policy. This creates operational gaps: Everruns may allow a route that OpenRouter's workspace budget or tier restrictions will reject; operators can't tell from within Everruns why a request was blocked upstream. This is EVE-566 in the OpenRouter follow-up sequence.

## How

**`inspect_openrouter_workspace`** — calls `/api/v1/auth/key`, converts OpenRouter's millionths-of-USD integers to float USD, and returns `OpenRouterKeyInfo`:
- `label`, `usage_usd`, `limit_usd`, `remaining_usd`, `is_free_tier`, `rate_limit`
- Raw API key is never included in the response

**`check_openrouter_policy_compatibility`** — fetches workspace info then runs `detect_policy_drift()` against caller-supplied routing params:
- `min_remaining_budget_usd` — threshold; drift reported if remaining < threshold
- `requires_paid_features` — drift reported when workspace is free-tier

Returns `PolicyCompatibilityReport { compatible, workspace, drifts }` where each `WorkspacePolicyDrift` is one of:
- `budget_exhausted` — spend cap reached
- `budget_below_threshold` — remaining budget below requested threshold
- `free_tier_restriction` — paid-tier features requested on free key

Capability registered in `CapabilityRegistry::with_builtins_for_grade()`.

## Risk

- Low — read-only; no writes to OpenRouter or any Everruns state
- Additive — new capability; no existing paths changed

## Security

Reviewed: TM-LLM, TM-API, TM-DOS

- `TM-LLM`: API key sourced from `provider_credential_store` and never returned to callers or logged. The tool response contains only derived metadata (usage, limits, tier, rate-limit).
- `TM-API`: Only reads `/api/v1/auth/key`; no write operations against the OpenRouter workspace. No new Everruns API endpoints added.
- `TM-DOS`: Single HTTP call per tool invocation; no loops. `reqwest::Client::new()` uses default timeout (no unbounded wait).

## Follow-ups

- Surface workspace policy info in the UI provider settings panel (deferred — requires server/API changes).
- Periodically cache workspace policy alongside the provider credential to avoid repeated API calls (deferred — no caching infrastructure for tool-context data yet).

## Checklist
- [x] Tests added or updated (15 unit tests covering parsing, drift detection, serialization, edge cases)
- [x] Backward compatibility considered (additive; no existing behavior changed)
- [x] Security review performed against relevant threat model categories
- [x] Final CI ran checks affected by this PR
- [x] All review comments addressed (code change or written reasoning)

---
_Generated by [Claude Code](https://claude.ai/code/session_014zWDagYxaty9q19wfeA3Hw)_

### CODE REVIEW

File: crates/core/src/capabilities/openrouter_workspace.rs [hunk 1]
```diff
@@ -0,0 +1,704 @@
...
+            Ok(k) => k,
+            Err(e) => return ToolExecutionResult::tool_error(e),
+        };
+
+        let client = reqwest::Client::new();
+        let response = match client
+            .get("https://openrouter.ai/api/v1/auth/key")
+            .bearer_auth(&api_key)
+            .send()
+            .await
```
[the remark this answers is not in the document]
Comment (chaliy) [reply crates/core/src/capabilities/openrouter_workspace.rs]: Fixed in the follow-up commit. Both `inspect_openrouter_workspace` and `check_openrouter_policy_compatibility` now check `context.network_access.is_url_allowed` before making any outbound call (returning a tool error if blocked), and build their HTTP client with `reqwest::Client::builder().timeout(Duration::from_secs(15))` instead of using the default client.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: crates/core/src/capabilities/openrouter_workspace.rs [hunk 2]
```diff
@@ -0,0 +1,704 @@
...
```
[the remark this answers is not in the document]
Comment (chaliy) [reply crates/core/src/capabilities/openrouter_workspace.rs]: Fixed — same as the `inspect_openrouter_workspace` fix. Both tools now enforce the network access allow-list and use a 15-second request timeout.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: crates/core/src/capabilities/openrouter_workspace.rs [hunk 3]
```diff
@@ -0,0 +1,704 @@
...
+        let is_free_tier = data
+            .get("is_free_tier")
+            .and_then(|v| v.as_bool())
+            .unwrap_or(false);
+
+        let rate_limit = data.get("rate_limit").and_then(|rl| {
+            let requests = rl.get("requests")?.as_u64()? as u32;
+            let interval = rl.get("interval")?.as_str()?.to_string();
+            Some(OpenRouterRateLimit { requests, interval })
+        });
```
[the remark this answers is not in the document]
Comment (chaliy) [reply crates/core/src/capabilities/openrouter_workspace.rs:128]: Fixed. Changed `as u64` → `try_into().ok()?` so an out-of-range rate-limit count returns `None` (rate_limit field is omitted from the result) rather than silently truncating.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: crates/core/src/capabilities/openrouter_workspace.rs [hunk 4]
```diff
@@ -0,0 +1,704 @@
...
+         Returns structured policy metadata without exposing the API key itself. \
+         Use this to understand workspace constraints before configuring model routing."
+    }
+
+    fn parameters_schema(&self) -> Value {
+        json!({
+            "type": "object",
+            "properties": {}
+        })
+    }
```
[the remark this answers is not in the document]
Comment (chaliy) [reply crates/core/src/capabilities/openrouter_workspace.rs:254]: Fixed. Added `"additionalProperties": false` to the `inspect_openrouter_workspace` parameter schema.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: crates/core/src/capabilities/openrouter_workspace.rs [hunk 5]
```diff
@@ -0,0 +1,704 @@
...
+                "requires_paid_features": {
+                    "type": "boolean",
+                    "description": "Set true if the routing config uses paid-tier features such as \
+                                    ZDR endpoints, data-collection controls, or paid-only providers. \
+                                    A drift is reported when the workspace is on the free tier.",
+                    "default": false
+                }
+            }
+        })
+    }
```
[the remark this answers is not in the document]
Comment (chaliy) [reply crates/core/src/capabilities/openrouter_workspace.rs:376]: Fixed. Added `"additionalProperties": false` and `"minimum": 0` to the `check_openrouter_policy_compatibility` schema. Also added a `.filter(|&v| v >= 0.0)` guard in `execute_with_context` so negative values passed through JSON (before schema validation) are silently treated as absent.

---
_Generated by [Claude Code](https://claude.ai/code)_
